python_modules/dagster/dagster_tests/core_tests/test_metadata.py

```diff
@@ -105,3 +105,8 @@ def test_table_schema_metadata_value():
 def test_json_metadata_value():
     assert JsonMetadataValue({"a": "b"}).data == {"a": "b"}
     assert JsonMetadataValue({"a": "b"}).value == {"a": "b"}
+
+
+def test_serdes_json_metadata():
+    old_bad_event_str = '{"__class__": "JsonMetadataEntryData", "data": {"float": {"__class__": "FloatMetadataEntryData", "value": 1.0}}}'
+    assert deserialize_value(old_bad_event_str, JsonMetadataValue)
```

Comment:
Potentially worth writing a test that goes back and forth a couple times.

Reply:
done